c_glib/arrow-dataset-glib/scanner.cpp

```diff
@@ -265,7 +265,7 @@ gad_scan_options_class_init(GADScanOptionsClass *klass)
   gobject_class->set_property = gad_scan_options_set_property;
   gobject_class->get_property = gad_scan_options_get_property;
 
-  auto scan_options = arrow::dataset::ScanOptions::Make(arrow::schema({}));
+  auto scan_options = std::make_shared<arrow::dataset::ScanOptions>();
```

Comment:
@mrkn https://issues.apache.org/jira/browse/ARROW-11782

Reply:
OK, I will create a new pull-request to remove the bindings of these internal classes.